src/diffpy/utils/diffraction_objects.py

```diff
@@ -36,26 +36,67 @@ def _setter_wmsg(attribute):
 
 class DiffractionObject:
     def __init__(
-        self, name=None, wavelength=None, scat_quantity=None, metadata=None, xarray=None, yarray=None, xtype=None
+        self,
+        xarray,
+        yarray,
+        xtype,
+        wavelength,
+        scat_quantity="",
```

Comment:
See discussion in issue about wavelength.

Reply:
done